### ISSUE TITLE
Plugin not working in ios platform : ionic

### ISSUE THREAD

**Issue by krithikuntukad** [Beaconstac/cordova-plugin-nearbee#1]
When adding platform ios to project it gives following error

"framework" tag with type "podspec" is deprecated and will be removed. Please use the "podspec" tag.
Running command: pod install --verbose
warning: Insecure world writable dir ..in PATH, mode 040777
pod: Command failed with exit code 1 Error output:

**Comment by sachinmobstac** [comment]
Please update iOS platform to 5.0.0 as per the requirements in the ReadMe.

**Comment by krithikuntukad** [comment]
I have updated iOS platform to 5.0.0 .i.e  i am using cordova ios 5.0.0 version still i am facing same issue

**Comment by krithikuntukad** [comment]
The exact error is :
Failed to install 'cordova-plugin-nearbee': Error: pod: Command failed with exit code 1
    at ChildProcess.whenDone (/Users/xxx/Downloads/cordova-plugin-nearbee-master/example/node_modules/cordova-common/src/superspawn.js:125:23)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Socket.stream.socket.on (internal/child_process.js:346:11)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at Pipe._handle.close [as _onclose] (net.js:554:12)
Failed to restore plugin "cordova-plugin-nearbee" from config.xml. You might need to try adding it again. Error: Error: pod: Command failed with exit code 1

**Comment by sachinmobstac** [comment]
@krithikuntukad 
Can you please confirm if you are missing the below pre-requisite?
**Pod iOS platform version should be 10.0** which requires the below tags in config.xml.

``` xml
<platform name="ios">
       <preference name="deployment-target" value="10.0" />
       ...
   </platform>
```

**Comment by krithikuntukad** [comment]
@sachinmobstac Yes.. The config.xml has the specified tag with same value..Still same error.

**Comment by sachinmobstac** [comment]
@krithikuntukad, The minimum required platform version for NearBee is 10.0. Can you tell me what is the platform version in the Podfile?

**Comment by krithikuntukad** [comment]
This is my podfile:
# DO NOT MODIFY -- auto-generated by Apache Cordova

platform :ios, '10.0'

target 'MyApp' do
	project 'MyApp.xcodeproj'

end


platform version is 10.0

**Comment by sachinmobstac** [comment]
@krithikuntukad, Have you installed the CocoaPods on your machine?

Please run the below commands and try again.

```
sudo gem install cocoapods
pod repo update
```

**Comment by krithikuntukad** [comment]
YEs i have them installed already.. I tied them .I will try one more time.

**Comment by krithikuntukad** [comment]
Still same error. Does anyone could integrate the plugin in their Ionic app and get it working successfull?..

**Comment by sachinmobstac** [comment]
@krithikuntukad, The NearBee SDK has dependencies on other Swift frameworks. Please add the below lines as well to the config.xml and try again. Make sure the CocoaPod version is 1.6.0 or higher.

``` xml
<platform name="ios">
       <preference name="SwiftVersion" value="4.2" />
       ...
   </platform>
```

**Comment by sachinmobstac** [comment]
@krithikuntukad, Did you try the above steps?

**Comment by krithikuntukad** [comment]
@sachinmobstac,
Now i can add plugin to project but building the application is throwing error:


Error: pod -1.6.1
Ionic-3.20.0
Cordova-ios:5.0.0

I also tried pod install in platforms/ios folder 


The following build commands failed:
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Ack/SocketAckEmitter.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Ack/SocketAckManager.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketAnyEvent.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngine.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineClient.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEnginePacketType.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEnginePollable.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineSpec.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineWebsocket.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketEventHandler.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketExtensions.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClient.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientConfiguration.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientOption.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientSpec.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOStatus.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketLogger.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Manager/SocketManager.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Manager/SocketManagerSpec.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Parse/SocketPacket.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Parse/SocketParsable.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketRawView.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketStringReader.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketTypes.swift
	CompileSwift normal x86_64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SSLSecurity.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(26 failures)
xcodebuild: Command failed with exit code 65
[ERROR] An error occurred while running subprocess cordova.
        
        cordova build ios exited with exit code 65.
        
        Re-running this command with the --verbose flag may provide more 
        information.

**Comment by sachinmobstac** [comment]
@krithikuntukad, You cannot run the build on the Simulator as the NearBee uses the CoreBluetooth which is always disabled in the simulator. You need to run the project on the device.

**Comment by krithikuntukad** [comment]
Okay  Its my mistake .Sorry ...Thanks  I will try building with device.

**Comment by krithikuntukad** [comment]
Building for device gives following error:

<unknown>:0: error: module map file '/Users/xxxxx/Library/Developer/Xcode/DerivedData/MyApp-afntfufeawwgjagsqzgsexjzrdkb/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/BuildProductsPath/Debug-iphoneos/Starscream/Starscream.modulemap' not found
<unknown>:0: error: module map file '/Users/xxxxx/Library/Developer/Xcode/DerivedData/MyApp-afntfufeawwgjagsqzgsexjzrdkb/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/BuildProductsPath/Debug-iphoneos/Starscream/Starscream.modulemap' not found
<unknown>:0: error: missing required module 'SwiftShims'

** ARCHIVE FAILED **


The following build commands failed:
	CompileSwiftSources normal armv7 com.apple.xcode.tools.swift.compiler
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Ack/SocketAckEmitter.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Ack/SocketAckManager.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketAnyEvent.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngine.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineClient.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEnginePacketType.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEnginePollable.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineSpec.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineWebsocket.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketEventHandler.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketExtensions.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClient.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientConfiguration.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientOption.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientSpec.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOStatus.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketLogger.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Manager/SocketManager.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Manager/SocketManagerSpec.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Parse/SocketPacket.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Parse/SocketParsable.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketRawView.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketStringReader.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketTypes.swift
	CompileSwift normal armv7 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SSLSecurity.swift
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Ack/SocketAckEmitter.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Ack/SocketAckManager.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketAnyEvent.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngine.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineClient.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEnginePacketType.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEnginePollable.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineSpec.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngineWebsocket.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketEventHandler.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketExtensions.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClient.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientConfiguration.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientOption.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOClientSpec.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketIOStatus.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketLogger.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Manager/SocketManager.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Manager/SocketManagerSpec.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Parse/SocketPacket.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Parse/SocketParsable.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Client/SocketRawView.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketStringReader.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SocketTypes.swift
	CompileSwift normal arm64 /Users/xxxxx/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Util/SSLSecurity.swift
(52 failures)
xcodebuild: Command failed with exit code 65
[ERROR] An error occurred while running subprocess cordova.
        
        cordova build ios --buildFlag=-UseModernBuildSystem=0 exited with exit 
        code 65.


Do I need to configure anything in xcode after adding platform ios?

**Comment by sachinmobstac** [comment]
@krithikuntukad, Can you add a dummy Swift file to the workspace and check again?

**Comment by krithikuntukad** [comment]
Sure .. Added dummy swift file to workspace and rebuilt .still same error...and xCode version I am using  is: 10.2.1

**Comment by sachinmobstac** [comment]
@krithikuntukad, Can you set the **Always Embed Swift Standard Libraries** to true in the workspace and try again?

**Comment by krithikuntukad** [comment]
Always Embed Swift Standard Libraries is set to true already..

**Comment by sachinmobstac** [comment]
@krithikuntukad, We got those errors, apparently the Cordova build command is failing. You can you open the xcworkspace and build and run the app using Xcode instead of cordova commands.

**Comment by krithikuntukad** [comment]
@sachinmobstac , With xCode I got following error,
I am guessing framework is built with different swift version than the swift version i am using.

/Users/sonatamobility/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/NearBee/NearBee/libNearBee.a(NearBeeBeacon+CoreDataClass.o) compiled with older version of Swift language (unknown ABI version 0x06) than previous files (unknown ABI version 0x07) file '/Users/sonatamobility/Desktop/Archana/Ionic3Projects/trainingApp/test/myHelloWorld/platforms/ios/Pods/NearBee/NearBee/libNearBee.a' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by krithikuntukad** [comment]
@sachinmobstac ,

_I rebuilt the frameworks with xCode 10.2.1,then
I have downloaded the cordova nearbee plugin._
I removed  line "<framework src="NearBee" type="podspec" ..>"  and added following 
  <source-file src="src/ios/framework/SocketIO.framework" target-dir="lib" framework="true" />
        <framework src="src/ios/framework/SocketIO.framework" embed="true" custom="true" />
          <source-file src="src/ios/framework/EddystoneScanner.framework" target-dir="lib" framework="true" />
        <framework src="src/ios/framework/EddystoneScanner.framework" embed="true" custom="true" />
          <source-file src="src/ios/framework/Starscream.framework" target-dir="lib" framework="true" />
        <framework src="src/ios/framework/Starscream.framework" embed="true" custom="true" />
          <source-file src="src/ios/framework/NearBee.framework" target-dir="lib" framework="true" />
        <framework src="src/ios/framework/NearBee.framework" embed="true" custom="true" />

_also copied all frameworks to src/ios/framework folder. and finally added a hook to link the frameworks._

_I got Success for "ionic cordova build ios" but got following error_

No visible @interface for 'NearBee' declares the selector 'displayContentOfEddystoneUrl

_So i only commented that particular function(NearbeePlugin.prototype.launchUrl) in plugin re built the code and Selected YES to always embed swift standard libraries.  
Used  swift version 5. Cordova ios 5.0.1  and finally I am able to call plugin methods except for commented function.._

Thanks a lot for support.. But it would be great help if u help me to fix following error "
No visible @interface for 'NearBee' declares the selector 'displayContentOfEddystoneUrl"

Thanks in advance

**Comment by sachinmobstac** [comment]
@krithikuntukad, I have released a new version which supports Swift 5.0. Please update, meanwhile I am working on the cordova build error.

**Comment by krithikuntukad** [comment]
@sachinmobstac,

Thank you

**Comment by krithikuntukad** [comment]
@sachinmobstac,
Error with latest plugin ,
When I build through xCode it is "'NearBee/NearBee-Swift.h' file not found"

and when I build using ionic cordova build ios command it is 
"** ARCHIVE FAILED **


The following build commands failed:
	CompileC /Users/xxxx/Library/Developer/Xcode/DerivedData/MyApp-afndwwdctanxhlerhmdmyujevdws/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/MyApp.build/Debug-iphoneos/MyApp.build/Objects-normal/arm64/NearbeePlugin.o /Users/xxxxx/Desktop/projects/Ionic3Projects/trainingApp/test/newPlugin/newPlugin/platforms/ios/MyApp/Plugins/cordova-plugin-nearbee/NearbeePlugin.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
xcodebuild: Command failed with exit code 65
[ERROR] An error occurred while running subprocess cordova.
        
        cordova build ios exited with exit code 65."

**Comment by sachinmobstac** [comment]
@krithikuntukad, Can you tell me the version number?

**Comment by krithikuntukad** [comment]
plugin version-0.0.7
ionic version-3.20.0
ionic cordova -5.0.1

**Comment by sachinmobstac** [comment]
@krithikuntukad, It works fine for me.

**Comment by krithikuntukad** [comment]
@sachinmobstac , I will recheck...

**Comment by krithikuntukad** [comment]
@sachinmobstac , 
Following are the versions:
cordova-plugin-nearbee ---0.0.7
cordova-ios ---5.0.1
I have added 2 lines in config xml
  <preference name="SwiftVersion" value="5" />
  <preference name="deployment-target" value="10.0" />
 
For me still it is the same error -
When I build through xCode it is "'NearBee/NearBee-Swift.h' file not found"

and when I build using ionic cordova build ios command it is
"** ARCHIVE FAILED **

The following build commands failed:
CompileC /Users/xxxx/Library/Developer/Xcode/DerivedData/MyApp-afndwwdctanxhlerhmdmyujevdws/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/MyApp.build/Debug-iphoneos/MyApp.build/Objects-normal/arm64/NearbeePlugin.o /Users/xxxxx/Desktop/projects/Ionic3Projects/trainingApp/test/newPlugin/newPlugin/platforms/ios/MyApp/Plugins/cordova-plugin-nearbee/NearbeePlugin.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
xcodebuild: Command failed with exit code 65
[ERROR] An error occurred while running subprocess cordova.

    cordova build ios exited with exit code 65."

**Comment by sachinmobstac** [comment]
Hi @krithikuntukad, Please follow the below steps and let me know if you are still getting the same issue:

1. Start a new ionic project
2. Modify the config.xml to include the below preferences
```xml
<platform name="ios">
       <preference name="deployment-target" value="10.0" />
       <preference name="SwiftVersion" value="5.0" />
       ...
   </platform>
```
Note: **You should be on the latest Xcode Version.**
3. `ionic cordova platforms add ios`
4. `ionic cordova plugin add cordova-plugin-nearbee`
5. Open the Project.xcworkspace file and run the project.

**Comment by krithikuntukad** [comment]
@sachinmobstac ,
I followed the steps u have specified earlier and i am still getting same issues.
"'NearBee/NearBee-Swift.h' file not found NearbeePlugin.m"

**Comment by sachinmobstac** [comment]
@krithikuntukad, To expedite the process can you please write to at support@mobstac.com and we will take it further.

**Comment by krithikuntukad** [comment]
@sachinmobstac , Sure.. Thank you...  and also i have a query related with the cordova plugin for android . It is working fine on android device ,but when I get notification in background mode and I click on notification ,notification will be opened in browser but on click of close icon user is not navigated back to the application . Do you have any idea why this behaviour? Since it is not push notification it is not handled by Android inbuilt service. Do i have to handle it in my Ionic code?

**Comment by sachinmobstac** [comment]
@krithikuntukad, The master version of the pod repo wasn't updated to the latest, which caused the issue on iOS. 

The above mentioned Android bug is fixed in the new version 0.0.8.

**Comment by krithikuntukad** [comment]
@sachinmobstac  Thank you.. I will install latest android cordova plugin and recheck.
